### PR TITLE
add new standard : react-native/package-dependencies/handle-dependenc…

### PR DESCRIPTION
**Pull request by justinemignot**
…ies-with-yarn-override

### CODE REVIEW

**Comment by bam-bot-tech** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  **<a href="https://github.com/bamlab/dev-standards/blob/new-standard-react-native-package-dependencies-with-yarn-override/react-native/package-dependencies/handle-dependencies-with-yarn-override.mo.md">handle-dependencies-with-yarn-override.mo.md</a>**: Does not seem to be included in the root readme
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  **<a href="https://github.com/bamlab/dev-standards/blob/new-standard-react-native-package-dependencies-with-yarn-override/react-native/package-dependencies/handle-dependencies-with-yarn-override.mo.md">handle-dependencies-with-yarn-override.mo.md</a>**: Does not seem to be included in the root summary
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: react-native/package-dependencies/handle-dependencies-with-yarn-override.mo.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+## Owner: [Justine Mignot](https://github.com/justinemignot)
+
+> **Note**: Please create an [issue](https://github.com/bamlab/dev-standards/issues/new) or even a Pull Request with your feedback, typo correction.
+
+## Context
+
+During this standard, we will go through how to handle versions dependencies of different packages to a same package.
+
+If some package doesn't use same versions of a package, depending of the order it will be called, it won't be the same version of package used. We want to ensure that the same version is used everywhere in the app. Normally this would require to manually edit `yarn.lock` file, but by chance, yarn has a feature override, which allows us to fix those versions dependencies.
```
Comment (tychota) [inline react-native/package-dependencies/handle-dependencies-with-yarn-override.mo.md]: Why is editing yarn.lock bad => deleted when installing new package or running yarn commands

File: react-native/package-dependencies/handle-dependencies-with-yarn-override.mo.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+We will go through the example of project which already contains a package (`moment-timezone`) which depends of package `moment` and in which we want to add `react-native-datepicker` which also depends on `moment`.
+
+## Prerequisites
+
+* You have 2 dependencies (Y1 et Y2) that depend on different versions of the same package (X). You can check that in step 1.
```
Comment (tychota) [inline react-native/package-dependencies/handle-dependencies-with-yarn-override.mo.md]: - Have a react native app installed
- Install moment-timezone (version"x.y.z")
- Install datepicker (version"x.y.z")

File: react-native/package-dependencies/handle-dependencies-with-yarn-override.mo.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+## Steps
+
+### Figure out a problem
+
+* If one of the packages (package X) you are using in your app is not working anymore (no particular error message but for example, in my case dates in the app appear in English whereas it was in French before, and phone was set up in French), it can be because of package dependencies. In my case, `moment.locale()` was not working anymore.
```
Comment (tychota) [inline react-native/package-dependencies/handle-dependencies-with-yarn-override.mo.md]: More bullet points:

- identify the case:
  - there is a regression in the app
  - you just install a new package
- then open yarn.lock
- look for your newly installed package
- look for this package deps (examples)
- try to find other version of each deps

**Review by tychota — APPROVED** [review]

File: react-native/package-dependencies/handle-dependencies-with-yarn-override.mo.md (hunk 3 above)
Comment (louiszawadzki) [reply react-native/package-dependencies/handle-dependencies-with-yarn-override.mo.md]: Is there something that exists to detect potential breakages? (maybe in v2 of this ;) )

**Pull request merged**